### ISSUE TITLE
[Comments] - Flakes: NixOS and Home Manager migration

### ISSUE THREAD

**Issue by gvolpe** [gvolpe/blog#16]

**Comment by Awan** [comment]
You can run home-manager switch --flake directly from any where and it will switch as well.

**Comment by Awan** [comment]
[![asciicast](https://asciinema.org/a/513631.svg)](https://asciinema.org/a/513631)

**Comment by gvolpe** [comment]
@Awan I use `nix build` directly these days 😉 https://github.com/gvolpe/nix-config/blob/master/switch#L7

**Comment by Awan** [comment]
![Just a new installation using flakes from my nix repo mithu branch](https://user-images.githubusercontent.com/42554663/184080754-af92a5a8-eb57-404d-ae41-96dd005687c9.jpg)

**Comment by gvolpe** [comment]
`nixos-install --flake` is very cool for fresh installations :) 

I haven't had the need to perform one in a long time (only have one working laptop right now), but definitely something I'd recommend doing. This blog post is probably a bit dated, as that was written when I first migrated to flakes...

**Comment by Awan** [comment]
yeah, its cool. I tried it yesterday on my laptop as I was moving to unstable. It worked like a charm. All I had to do is clone my [repo](https://github.com/Awan/nix), create a new branch, stripped down it for fast installation and then pushed back to gh. I just added my encrypted drive UUID and then `nixos-install --flake github:Awan/nix/mithu#x1c`. mithu is the branch if you don't want to use default master/main branch...

On 11/08, Gabriel Volpe wrote:
>`nixos-install --flake` is very cool for fresh installations :)
>
>I haven't had the need to perform one in a long time (only have one working laptop right now), but definitely something I'd recommend doing. This blog post is probably a bit dated, as that was written when I first migrated to flakes...
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/gvolpe/blog/issues/16#issuecomment-1211650488
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>


Abdullah Khabir

Cyber Security Analyst

https://abdullah.today

C20F 2707 3025 2569 BAC5
534B 7820 6670 C19D 1580

**Comment by Awan** [comment]
[Here](https://github.com/gvolpe/nix-config/blob/2abcb0f8684324e21b29398f9c4a6fb2a097a0a6/switch#L7) you can use `sudo -E nixos-rebuild --flake .#hostname` and it will save you from activation next step ;-)

**Comment by gvolpe** [comment]
@Awan I think you got the wrong line, you're pointing to the Home Manager activation, you don't need `sudo` for that.

**Comment by Awan** [comment]
@gvolpe yeah, I just noticed it. That `rebuild_system` is what I was mentioning.

**Comment by gvolpe** [comment]
There's no activation there, though 🤔 It's just `sudo nixos-rebuild switch --flake .#tongfang-amd`
